### PR TITLE
Remove incompatible versions in upgrade manifest

### PR DESCRIPTION
**Pull request by jacek-lewandowski**

### CODE REVIEW

File: upgrade_tests/upgrade_manifest.py [hunk 1]
```diff
@@ -248,14 +248,16 @@ def build_upgrade_pairs():
...
                     destination_meta = newmeta
 
-            valid_upgrade_pairs.append(
-                UpgradePath(
-                    name=path_name,
-                    starting_version=origin_meta.version,
-                    upgrade_version=destination_meta.version,
-                    starting_meta=origin_meta,
-                    upgrade_meta=destination_meta
+            from upgrade_tests.upgrade_through_versions_test import jdk_compatible_steps
```
Comment (michaelsembwever) [inline upgrade_tests/upgrade_manifest.py]: nit: maybe `jdk_compatible_steps` should be moved here ( to upgrade_manifest.py ).  this is more the base file.

**Review by michaelsembwever — APPROVED** [review]
good catch 👍

File: upgrade_tests/upgrade_manifest.py (hunk 1 above)
Comment (jacek-lewandowski) [reply upgrade_tests/upgrade_manifest.py]: sure, I can move it

**Comment by jacek-lewandowski** [comment]
Wow https://ci-cassandra.apache.org/view/patches/job/Cassandra-devbranch-dtest-upgrade/1744/ - all green :)

**Comment by jacek-lewandowski** [comment]
Running CircleCI as well: https://app.circleci.com/pipelines/github/jacek-lewandowski/cassandra/825/workflows/cfeae013-ae88-41f0-88a7-309dbdac3d7b

**Pull request merged**